multik-jvm/src/test/kotlin/org/jetbrains/kotlinx/multik_jvm/linAlg/JvmLinAlgTest.kt

```diff
@@ -0,0 +1,254 @@
+
+        assertEquals(10.0, mk.linalg.norm(d2arrayLong1, 1))
+        assertEquals(10.0, mk.linalg.norm(d2arrayLong2, 1))
+
+    }
+
+//    @Test
+//    fun `test solveTriangleInplace`() {
+//        val procedurePrecision = 1e-5
+//
+//        val rnd = Random(123)
+//        val n = 18
+//        val m = 100
+//
+//        val a = mk.d2array<Double>(n, n) {0.0}
+//        for (i in 0 until n) {
+//            for (j in 0 .. i) {
+//                if (j == i) {
+//                    a[i, j] = 1.0
+//                } else {
+//                    a[i, j] = rnd.nextDouble()
+//                }
+//            }
+//        }
+//
+//        val b = mk.d2array<Double>(n, m) {0.0}
+//        for (i in 0 until n)
+//            for (j in 0 until m)
+//                b[i, j] = rnd.nextDouble()
+//
+//        val saveB = b.deepCopy()
+//
+//        JvmLinAlg.solveTriangleInplace(a, 0, 0, n, b, 0, 0, m)
+//
+//        //check equality up to precision
+//        assert(max(dot(a, b) - saveB) < procedurePrecision && min(dot(a, b) - saveB) > -procedurePrecision)
+//    }
+
+//    @Test
+//    fun `gemm test`() {
+//        val procedurePrecision = 1e-5
+//
+//        val a = mk.ndarray(mk[mk[1.0, 2.0, 3.0, 2.0, 1.0], mk[4.0, 5.0, 6.0, 5.0, 4.0]])
+//        val b = mk.ndarray(mk[mk[7.0, 8.0], mk[9.0, 10.0], mk[11.0, 12.0], mk[13.0, 14.0], mk[15.0, 16.0]])
+//        val c = mk.ndarray(mk[mk[11.0, 13.0], mk[15.0, 17.0]])
+//        val adotb = mk.ndarray(mk[mk[110.0, 121.0], mk[279.0, 305.0]])
+//
+//        //        proof: numpy
+//        //
+//        //        a = np.array([[1.0, 2.0, 3.0, 2.0, 1.0], [4.0, 5.0, 6.0, 5.0, 4.0]])
+//        //        b = np.array([[7.0, 8.0], [9.0, 10.0], [11.0, 12.0], [13.0, 14.0], [15.0, 16.0]])
+//        //        print(a @ b + c)
+//
+//        gemm(a, 0, 0, 2, 5, 1.0,
+//             b, 0, 0, 5, 2,
+//             c, 0, 0, 2, 2, 1.0)
+//
+//        assert(c == adotb)
+//
+//    }
+
+
+
+    @Test
+    fun `test plu`() {
+
+        val procedurePrecision = 1e-5
+
+        val iters = 10000
+        val sideFrom = 1
+        val sideUntil = 100
+        for (all in 0 until iters) {
+
+            val rnd = Random(System.currentTimeMillis())
+            val m = rnd.nextInt(sideFrom, sideUntil)
+            val n = rnd.nextInt(sideFrom, sideUntil)
+
+
+            val a = mk.d2array<Double>(m, n) { rnd.nextDouble() }
+
+            val (P, L, U) = PLU(a)
+
+            val diff = a - dot(P, dot(L, U))
+            val maxdiff = max(diff)
+            val mindiff = min(diff)
+
+            val ok = maxdiff < procedurePrecision && maxdiff - mindiff < procedurePrecision
+
+
+            if(!ok) {
+                println("wrong on:")
+                println(a)
+                println("PLU gives:")
+                println(P)
+                println(L)
+                println(U)
+                println("P*L*U = ")
+                println(dot(P, dot(L, U)))
+                assert(false)
+
+            }
```

Comment:
It is better to use the standard asserts from kotlin.test, you can put the necessary information in the message.